plugins/modules/mount.py

```diff
@@ -250,13 +255,14 @@ def mount(module):
     else:
         mount_dir = module.params['mount_dir']
         mount_over_dir = module.params['mount_over_dir']
-        if mount_dir or mount_over_dir:
-            if is_fspath_mounted(module, mount_dir, mount_over_dir):
-                if mount_dir:
-                    result['msg'] = "Filesystem/Mount point '%s' already mounted" % mount_dir
-                else:
-                    result['msg'] = "Filesystem/Mount point '%s' already mounted" % mount_over_dir
-                return
+        if is_fspath_mounted(module, mount_dir, mount_over_dir):
+            # if both mount_dir and mount_over_dir is given then check for
+            # mount_over_dir
+            if (mount_dir and mount_over_dir) or mount_over_dir:
```

Comment:
Same here, should be checking mount_over_dir only sufficient?

Reply:
yes, you are right